rewrite-core/src/main/java/org/openrewrite/Recipe.java

```diff
@@ -72,8 +73,8 @@ public String getJacksonPolymorphicTypeTag() {
     private static final TreePrinter<ExecutionContext> MARKER_ID_PRINTER = new TreePrinter<ExecutionContext>() {
         @Override
         public void doBefore(@Nullable Tree tree, StringBuilder printerAcc, ExecutionContext executionContext) {
-            if (tree != null) {
-                String markerIds = tree.getMarkers().entries().stream()
+            if (tree instanceof Markable) {
+                String markerIds = ((Markable)tree).getMarkers().entries().stream()
```

Comment:
Are you suggesting that things that are currently `Markable` should, instead of a collection of Markers, have only a single Marker? If only one Marker is allowed on an AST element then that Marker would have to, internally at least, be a collection of Markers... in which case what's the advantage of getting rid of Markable?